subscriber.go

```diff
@@ -811,11 +888,77 @@ func (h *handler) handle(ctx context.Context, nextCid cid.Cid, sel ipld.Node, wr
 		return nil, nil
 	}
 
-	err := syncer.Sync(ctx, nextCid, sel)
-	if err != nil {
-		return nil, err
+	var limitOK bool
+	var origLimit selector.RecursionLimit
+	// Only attempt to extract original recursion limit from selector if segmentation is enabled
+	// and there is a segment block hook.
+	//
+	// Not that we need the action in segment block hook to know which CID to sync in next segment.
+	// Therefore, it has to be set for segmented sync to function correctly.
+	if segdl > 0 && segbh != nil {
+		origLimit, limitOK = getRecursionLimit(sel)
+	}
```

Comment:
What if the caller wants to sync in one go (no segments), but also wants a recursion depth limit?  Does this mean that in order to have limited depth, a `SegmentBlockHookFunc` is required?

Reply:
If the caller wants to sync in one go _and_ have a recursion depth the caller should use a selector with the desired recursion depth and disable segmentation.

Those two are separate things unless I am missing something?

Reply:
Note that we are _not adding_ recursion here; we simply modify it to max segment depth. In fact no segmented sync will be performed at all if the original selector doesn't have some recursion limit regardless of whether segmentation is enabled or not.